### PR TITLE
DEVXP-2784 - Add poetry

### PR DESCRIPTION
**Pull request by vschoener**
This PR adds poetry and do a refactoring with disutils and packaging https://github.com/pytorch/pytorch/issues/69894#issuecomment-993544244

Let me know if we need to do more change or iterate :)

Also, the tests are failing for test_libyear on master cause the requirement have changed since the las time. it feels a bit flaky btw.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+
+[tool.poetry.dependencies]
+python = "^3.12"
+requests = "^2.31.0"
+prettytable = "^3.10.0"
+python-dateutil = "^2.9.0.post0"
+setuptools = "^69.2.0"
+packaging = "^24.0"
+coverage = "^7.0.5"
```
Comment (vitorquintanilha) [inline pyproject.toml]: If I'm not wrong this is only used for testing? Can we move it to `dev`?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+[tool.poetry]
+name = "libyear"
+version = "0.1.0"
```
Comment (vitorquintanilha) [inline pyproject.toml]: I think we could keep the version from upstream? `0.2.1`?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+coverage = "^7.0.5"
+
+
+[tool.poetry.group.dev.dependencies]
+pytest = "^7.0.0"
+pytest-sugar = "^0.9.4"
+pytest-cov = "^4.0.0"
+pytest-asyncio = "^0.23.0"
+pytest-runner = "^6.0.1"
+
```
Comment (vitorquintanilha) [inline pyproject.toml:27]: We should add the requirements from `test_requirements.txt`  here and delete that file! (we are missing `pytest-vcr`).

**Review by vitorquintanilha — CHANGES_REQUESTED** [review]
Need to update the lock file with `poetry lock --no-update`. Also tests should work once we add `pytest-vcr` as a dev requirement.

File: pyproject.toml (hunk 3 above)
Comment (vschoener) [reply pyproject.toml:27]: Without pytest-vcr I was able to run the test anyway 🤔 , but I'm gonna add it

**Comment by vschoener** [comment]
> poetry lock --no-update

I would have been surprised if the test would work with this package ^^. I have the same remaining error, see the capture

image

**Comment by vschoener** [comment]
> > poetry lock --no-update
> 
> I would have been surprised if the test would work with this package ^^. I have the same remaining error, see the capture
> 
> image

Tests work by running `poetry run pytest` but not with the IDE (PyCharm)

**Pull request merged**